### ISSUE TITLE
Provide grid in backend template for collections

### ISSUE THREAD

**Issue by DanielSiepmann** [FriendsOfTYPO3/content-blocks#68]
Our use case:

We want nested content (simple grids). This is possible with only a short tweak to TYPO3 (I can share that in an additional issue if wanted). But we are not able to alter the data passed to the backend preview in order to build a Grid which can be rendered by TYPO3 (as done by EXT:container).

That's the only missing piece for us to provide proper and easy and fast previews for nested records from various tables.
I was not able to easily add this to the current code base. Feel free to contact me to show you the issues.

Maybe use an Event which offers the raw data, processed data and current GridColumnItem. Only allow to modify the processed data.
That way developers can build the necessary stuff.

**Comment by DanielSiepmann** [comment]
It might even be possible and better to auto generate the grid and pass it to the template.
This could be done for all columns of Type `Collection`?

**Comment by nhovratov** [comment]
If I understand you correctly you want to use the DataProcessor of EXT:container in the backend previews?

```
tt_content.b13-2cols-with-header-container {
    templateName = 2ColsWithHeader
    templateRootPaths {
        10 = EXT:container_example/Resources/Private/Templates
    }
    dataProcessing {
        100 = B13\Container\DataProcessing\ContainerProcessor
    }
}
```
But this doesn't work as there is no API/Event to hook into the backend preview generation of Content Blocks yet?

> It might even be possible and better to auto generate the grid and pass it to the template.
This could be done for all columns of Type Collection?

I don't understand what you mean here with Collections.

**Comment by DanielSiepmann** [comment]
The extension provides a field of type "Collection": https://docs.typo3.org/p/contentblocks/content-blocks/main/en-us/YamlReference/FieldTypes/Collection/Index.html

I don't want to mirror the frontend DataProcessing, but to allow custom processing of data, e.g. via an Event. Our concrete use case is a collection and I guess the custom processing is not necessary, but the extension could auto generate a grid for each collection.

That way each collection item can be rendered as proper grid within the backend preview. The same way as done by EXT:container for its child content elements. Just each collection would have a single column. That way, child elements like "Slides" of a slider content block or "Tabs" of a Tab content block will be rendered by TYPO3 natively which might improve the user experience.
We play around with this in one of our customer projects right now (still on v10). There we also added a collapse feature for those columns:
![image](https://github.com/nhovratov/content-blocks/assets/354250/fa6f80fc-15a5-42b0-8dbb-23df1b2b4758) 
![image](https://github.com/nhovratov/content-blocks/assets/354250/e3945fc9-d4bc-49d8-98ee-b0cb98be9731)

We still play around with this.

I can also share screenshot or a short screen share session if that helps.

**Comment by nhovratov** [comment]
This is also very nice, because hidden inline fields are displayed as such. Otherwise custom styling would be needed. And the control buttons are pure gold! From our session today, I saw that some adjustments in the core are needed. Could you summarize all adjustments for me again, so we could possibly fix this in the Core. I would love to see that!

**Comment by DanielSiepmann** [comment]
I guess we need the following within the core:
- Adjust the Grid classes to make the table more flexible. Right now some of the classes have "tt_content" hard coded. But the inline relations might refer to other record types as well. https://github.com/TYPO3/typo3/blob/main/typo3/sysext/backend/Classes/View/BackendLayout/Grid/GridColumnItem.php

Maybe some more things I'm not aware of yet.

And those things in general:
- Adjust the ContentFetcher (E.g. via event listener) to not fetch tt_content which is child of another tt_content)
- Adjust the TypoScript where condition to not fetch tt_content which is child of another tt_content)

**Comment by nhovratov** [comment]
:heavy_check_mark:   First patch addressing GridColumnItem: https://review.typo3.org/c/Packages/TYPO3.CMS/+/82411

**Comment by DanielSiepmann** [comment]
Further issues:
- GridColumn::isActive()
- GridColumn::isUnassigned()
- GridColumn::isUnused()
- I still get some unexpected output but am unable to find the cause. It should be GridColumn::getAfterSectionMarkup() but it doesn't return the value. `<f:format.raw>{column.afterSectionMarkup}</f:format.raw>`

**Comment by DanielSiepmann** [comment]
I pushed a PoC branch which results in the following:

```yaml
-
    identifier: 'children'
    type: 'Collection'
    minitems: 1
    foreign_table: 'tt_content'
    shareAcrossTables: true
    shareAcrossFields: true
```

![image](https://github.com/nhovratov/content-blocks/assets/354250/e7a820a8-802e-4b91-90b5-f07edd9aba5d)

**Comment by nhovratov** [comment]
>  GridColumn::isActive()
    GridColumn::isUnassigned()
    GridColumn::isUnused()
    I still get some unexpected output but am unable to find the cause. It should be GridColumn::getAfterSectionMarkup() but it doesn't return the value. <f:format.raw>{column.afterSectionMarkup}</f:format.raw>

Do you think these things are needed? getAfterSectionMarkup is only filled by an Event, or?
Have a look at my comment under the PR, I believe it is better to provide 1 working way to render these children.

**Comment by DanielSiepmann** [comment]
Those issues all seem to be resolved by your approach to have a custom Fluid Partial.
